modules/plugins/languages/tex/build/default.nix

```diff
@@ -0,0 +1,113 @@
+{
+  config,
+  lib,
+  ...
+}: let
+  inherit (lib.options) mkOption;
+  inherit (lib.types) str nullOr;
+  inherit (builtins) filter isAttrs hasAttr attrNames length elemAt;
+  inherit (lib.nvim.config) mkBool;
+
+  cfg = config.vim.languages.tex;
+
+  enabledBuildersCount = let
+    # This function will sort through the builder options and count how many
+    # builders have been enabled.
+    getEnabledBuildersCount = {
+      enabledBuildersCount ? 0,
+      index ? 0,
+      builderNamesList ? (
+        filter (
+          x: let
+            y = cfg.build.builders.${x};
+          in (isAttrs y && hasAttr "enable" y)
+        ) (attrNames cfg.build.builders)
+      ),
+    }: let
+      currentBuilderName = elemAt builderNamesList index;
+      currentBuilder = cfg.build.builders.${currentBuilderName};
+      nextIndex = index + 1;
+      newEnabledBuildersCount =
+        if currentBuilder.enable
+        then enabledBuildersCount + 1
+        else enabledBuildersCount;
+    in
+      if length builderNamesList > nextIndex
+      then
+        getEnabledBuildersCount {
+          inherit builderNamesList;
+          enabledBuildersCount = newEnabledBuildersCount;
+          index = nextIndex;
+        }
+      else newEnabledBuildersCount;
+  in (getEnabledBuildersCount {});
+in {
+  imports = [
+    ./builders
+  ];
+
+  options.vim.languages.tex.build = {
+    enable =
+      mkBool (
+        if enabledBuildersCount > 1
+        then throw "nvf-tex-language does not support having more than 1 builders enabled!"
+        else (enabledBuildersCount == 1)
+      ) ''
+        Whether to enable configuring the builder.
+
+        By enabling any of the builders, this option will be automatically set.
+        If you enable more than one builder then an error will be thrown.
+      '';
```

Comment:
I'd rather if you asserted this in `config.assertions` instead of this throw here.